### ISSUE TITLE
Won't build on standard OpenJDK 17 on Fedora 38

### ISSUE THREAD

**Issue by makingglitches** [VUE/VUE#62]
Looks like this incredibly useful project is unfortunately not updated for openjdk 17.0

Which is not surprising since I used this wayyyyy back in the early 2000s and it always seemed to work fine.

I imagine its a matter of installing the needed JRE and JDK to compile and run it side by side but that sounds extremely... dicey.

Anyway the  **ant compile** command is throwing numerous warnings but also some errors.

It fails out of the box, you have to update the **src/build.xml** variable **target.version** to a number of 7 or higher for javac to even run, at line 66 in the build rules file.

Once this is done the following extensive list of issues is popping up.

 [javac] warning: [options] bootstrap class path not set in conjunction with -source 7
    [javac] warning: [options] source value 7 is obsolete and will be removed in a future release
    [javac] warning: [options] target value 7 is obsolete and will be removed in a future release
    [javac] warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/gui/GUI.java:988: warning: as of release 9, '_' is a keyword, and may not be used as an identifier
    [javac]             } catch (Throwable _) {}
    [javac]                                ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/BrowseDataSource.java:31: warning: FileURLConnection is internal proprietary API and may be removed in a future release
    [javac] import sun.net.www.protocol.file.FileURLConnection;
    [javac]                                 ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VUE.java:17: warning: [removal] AppletContext in java.applet has been deprecated and marked for removal
    [javac] import java.applet.AppletContext;
    [javac]                   ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VUE.java:61: warning: [removal] JApplet in javax.swing has been deprecated and marked for removal
    [javac] import javax.swing.JApplet;
    [javac]                   ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VueApplet.java:32: warning: [removal] AccessController in java.security has been deprecated and marked for removal
    [javac] import java.security.AccessController;
    [javac]                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VueApplet.java:54: warning: XMLScanner is internal proprietary API and may be removed in a future release
    [javac] import com.sun.org.apache.xerces.internal.impl.XMLScanner;
    [javac]                                               ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/Util.java:2325: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]             return new Integer(tenX / 10).toString();
    [javac]                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/Util.java:2327: warning: [removal] Float(float) in Float has been deprecated and marked for removal
    [javac]             return new Float( ((float)tenX) / 10f ).toString();
    [javac]                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LWComponent.java:1020: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]     private static final Integer _DefaultInteger = new Integer(0);
    [javac]                                                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LWComponent.java:1031: warning: [removal] Integer(String) in Integer has been deprecated and marked for removal
    [javac]         void setBy(String s) { set(new Integer(s)); }
    [javac]                                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LWComponent.java:1034: warning: [removal] Float(float) in Float has been deprecated and marked for removal
    [javac]     private static final Float _DefaultFloat = new Float(0f);
    [javac]                                                ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LWComponent.java:1040: warning: [removal] Float(String) in Float has been deprecated and marked for removal
    [javac]         void setBy(String s) { set(new Float(s)); }
    [javac]                                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/filter/NodeFilter.java:177: warning: [removal] Integer(String) in Integer has been deprecated and marked for removal
    [javac]                 statement.setValue(new Integer(value.toString()));
    [javac]                                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LWLink.java:1373: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]         Object old = new Integer(mCurveControls);
    [javac]                      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VueAction.java:212: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]     private static final Boolean IS_NOT_A_TOGGLER = new Boolean(false);
    [javac]                                                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:677: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]             void init() { putValue("property.value", new Integer(0)); } // for use in a MenuButton
    [javac]                                                      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:688: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]             void init() { putValue("property.value", new Integer(1)); }
    [javac]                                                      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:699: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]             void init() { putValue("property.value", new Integer(2)); }
    [javac]                                                      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:1332: error: cannot find symbol
    [javac]       	      netscape.javascript.JSObject win = (netscape.javascript.JSObject) netscape.javascript.JSObject.getWindow(VueApplet.getInstance());
    [javac]       	                                                                                                    ^
    [javac]   symbol:   method getWindow(JApplet)
    [javac]   location: class JSObject
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:1334: warning: non-varargs call of varargs method with inexact argument type for last parameter;
    [javac]       	      win.call("doImportMap", arguments);
    [javac]       	                              ^
    [javac]   cast to Object for a varargs call
    [javac]   cast to Object[] for a non-varargs call and to suppress this warning
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:1352: error: cannot find symbol
    [javac]     				netscape.javascript.JSObject win = (netscape.javascript.JSObject) netscape.javascript.JSObject.getWindow(VueApplet.getInstance());
    [javac]     				                                                                                              ^
    [javac]   symbol:   method getWindow(JApplet)
    [javac]   location: class JSObject
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:1354: warning: non-varargs call of varargs method with inexact argument type for last parameter;
    [javac]   	      	      win.call("doImportUrl", arguments);
    [javac]   	      	                              ^
    [javac]   cast to Object for a varargs call
    [javac]   cast to Object[] for a non-varargs call and to suppress this warning
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:1360: error: cannot find symbol
    [javac]     				netscape.javascript.JSObject win = (netscape.javascript.JSObject) netscape.javascript.JSObject.getWindow(VueApplet.getInstance());
    [javac]     				                                                                                              ^
    [javac]   symbol:   method getWindow(JApplet)
    [javac]   location: class JSObject
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:1362: warning: non-varargs call of varargs method with inexact argument type for last parameter;
    [javac]   	      	      win.call("doImportFile", arguments);
    [javac]   	      	                               ^
    [javac]   cast to Object for a varargs call
    [javac]   cast to Object[] for a non-varargs call and to suppress this warning
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/UndoManager.java:382: warning: [removal] Float(float) in Float has been deprecated and marked for removal
    [javac]                 value = new Float(curValue.intValue() + inc * i);
    [javac]                         ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/UndoManager.java:393: warning: [removal] Double(double) in Double has been deprecated and marked for removal
    [javac]                 value = new Double(curValue.intValue() + inc * i);
    [javac]                         ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ds/Schema.java:1309: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]         	existingRows.put(values[matrixColNums.get(rowName)], new Integer(fromRow.mmap.size()));
    [javac]         	                                                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ds/Schema.java:1318: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]         		existingRows.put(values[matrixColNums.get(rowName)], new Integer(fromRow.mmap.size()));
    [javac]         		                                                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ds/Schema.java:1346: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]         		existingRows.put(values[matrixColNums.get(colName)], new Integer(toRow.mmap.size()));
    [javac]         		                                                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ds/Schema.java:1361: warning: [removal] Integer(String) in Integer has been deprecated and marked for removal
    [javac]     	int matrixSize = new Integer(ds.getMatrixSizeField()).intValue();
    [javac]     	                 ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/DrawContext.java:480: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]     public static final Boolean DEFAULT = new Boolean(false); // true/false not relevant
    [javac]                                           ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LinkTool.java:724: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]                 setterAction.putValue("property.value", new Integer(curveCount)); // this may be handy
    [javac]                                                         ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/gui/GUI.java:423: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]         if (DEBUG.Enabled) UIManager.put("ScrollBar.width", new Integer(64)); // just to know we've taken effect
    [javac]                                                             ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/MapViewer.java:8980: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]     private static final Boolean UNSET_BOOL = new Boolean(false);
    [javac]                                               ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/RichTextBox.java:1031: warning: [removal] Integer(String) in Integer has been deprecated and marked for removal
    [javac]        		Integer i = new Integer(a.toString());
    [javac]        		            ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/RichTextBox.java:1203: warning: [removal] Integer(String) in Integer has been deprecated and marked for removal
    [javac]     		Integer i = new Integer(a.toString());
    [javac]     		            ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/gui/DockWindow.java:1008: warning: SurfaceData is internal proprietary API and may be removed in a future release
    [javac]         sun.java2d.SurfaceData sd = null;
    [javac]                   ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/gui/DockWindow.java:1009: warning: SunGraphics2D is internal proprietary API and may be removed in a future release
    [javac]         if (g instanceof sun.java2d.SunGraphics2D)
    [javac]                                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/gui/DockWindow.java:1010: warning: SunGraphics2D is internal proprietary API and may be removed in a future release
    [javac]             sd = ((sun.java2d.SunGraphics2D)g).surfaceData;
    [javac]                              ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/preferences/implementations/WindowPropertiesPreference.java:128: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac] 		defaultEnabledVal = (new Boolean(box.isSelected()).toString());
    [javac] 		                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/preferences/implementations/WindowPropertiesPreference.java:171: warning: [removal] Boolean(String) in Boolean has been deprecated and marked for removal
    [javac] 		return new Boolean(set);//Boolean.valueOf((String)table.get(ENABLED_KEY));
    [javac] 		       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LWImage.java:941: warning: [removal] Double(double) in Double has been deprecated and marked for removal
    [javac]             public Object getValue(LWComponent c) { return new Double(((LWImage)c).getRotation()); }
    [javac]                                                            ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Images.java:3213: warning: ToolkitImage is internal proprietary API and may be removed in a future release
    [javac]             if (iconSource instanceof sun.awt.image.ToolkitImage) { // will probably always be a ToolkitImage
    [javac]                                                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Images.java:3217: warning: ToolkitImage is internal proprietary API and may be removed in a future release
    [javac]                 transparency = ((sun.awt.image.ToolkitImage)iconSource).getColorModel().getTransparency();
    [javac]                                               ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/preferences/implementations/ShowIconsPreference.java:287: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac] 			return new Boolean(true);
    [javac] 			       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/preferences/implementations/ShowIconsPreference.java:305: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac] 			return new Boolean(true);
    [javac] 			       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/preferences/implementations/ShowIconsPreference.java:339: warning: [removal] Boolean(String) in Boolean has been deprecated and marked for removal
    [javac] 		return new Boolean("true");
    [javac] 		       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/preferences/implementations/ShowIconsPreference.java:347: warning: [removal] Boolean(String) in Boolean has been deprecated and marked for removal
    [javac] 		return new Boolean("true");
    [javac] 		       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/BrowseDataSource.java:368: warning: FileURLConnection is internal proprietary API and may be removed in a future release
    [javac]         if (conn instanceof FileURLConnection)
    [javac]                             ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/filter/Key.java:41: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]             defaultValue = new Integer(0);
    [javac]                            ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/filter/Key.java:43: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]             defaultValue = new Boolean(true);
    [javac]                            ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VUE.java:141: warning: [removal] AppletContext in java.applet has been deprecated and marked for removal
    [javac]     private static AppletContext sAppletContext 	= null;
    [javac]                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VUE.java:707: warning: [removal] AppletContext in java.applet has been deprecated and marked for removal
    [javac]     public static void setAppletContext(AppletContext ac) {
    [javac]                                         ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VUE.java:710: warning: [removal] AppletContext in java.applet has been deprecated and marked for removal
    [javac]     public static AppletContext getAppletContext() {
    [javac]                   ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VUE.java:730: warning: [removal] AccessControlException in java.security has been deprecated and marked for removal
    [javac]             } catch (java.security.AccessControlException e) {
    [javac]                                   ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VUE.java:3751: warning: [removal] JApplet in javax.swing has been deprecated and marked for removal
    [javac]     		JApplet app =  VueApplet.getInstance();
    [javac]     		^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/PathwayPanel.java:803: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]             intArray[0] = new Integer(0);
    [javac]                           ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/PathwayPanel.java:804: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]             intArray[1] = new Integer(1);
    [javac]                           ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:123: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		fadeTable.put(new Integer(100), label100);
    [javac] 		              ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:124: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		fadeTable.put(new Integer(80), label80);
    [javac] 		              ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:125: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		fadeTable.put(new Integer(60), label60);
    [javac] 		              ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:126: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		fadeTable.put(new Integer(40), label40);
    [javac] 		              ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:127: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		fadeTable.put(new Integer(20), label20);
    [javac] 		              ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:128: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		fadeTable.put(new Integer(0), label0);
    [javac] 		              ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:162: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		depthTable.put(new Integer(0), labelOff);
    [javac] 		               ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:163: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		depthTable.put(new Integer(1), label1);
    [javac] 		               ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:164: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		depthTable.put(new Integer(2), label2);
    [javac] 		               ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:165: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		depthTable.put(new Integer(3), label3);
    [javac] 		               ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:166: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		depthTable.put(new Integer(4), label4);
    [javac] 		               ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/InteractionTools.java:167: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		depthTable.put(new Integer(5), label5);
    [javac] 		               ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ProviderListCellRenderer.java:102: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]     	checklist.set(index,new Boolean(true));    	
    [javac]     	                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ProviderListCellRenderer.java:136: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]         	checklist.add(index,new Boolean(false));
    [javac]         	                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ProviderListCellRenderer.java:247: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]     		checklist.set(i,new Boolean(false));
    [javac]     		                ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/action/SaveAction.java:198: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]                             row.add(new Boolean(true));
    [javac]                                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/action/SaveAction.java:200: warning: [removal] Long(long) in Long has been deprecated and marked for removal
    [javac]                             row.add(new Long(file.length()));
    [javac]                                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/RecentlyOpenedFilesManager.java:77: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		list.set(0, (Object)(new Integer(i).toString()));
    [javac] 		                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/compare/ui/WeightVisualizationSettingsPanel.java:325: warning: [removal] Double(double) in Double has been deprecated and marked for removal
    [javac]         boundaries.add(new Double((double)0.0 + (double)0.5));
    [javac]                        ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/compare/ui/WeightVisualizationSettingsPanel.java:329: warning: [removal] Double(double) in Double has been deprecated and marked for removal
    [javac]             boundaries.add(new Double((double)pi.getEnd()+(double)0.5));
    [javac]                            ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/compare/ui/WeightVisualizationSettingsPanel.java:348: warning: [removal] Double(double) in Double has been deprecated and marked for removal
    [javac]         boundaries.add(new Double((double)0.0 + (double).5));
    [javac]                        ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/compare/ui/WeightVisualizationSettingsPanel.java:352: warning: [removal] Double(double) in Double has been deprecated and marked for removal
    [javac]             boundaries.add(new Double((double)pi.getEnd())+(double).5);
    [javac]                            ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/RecentlyUsedColorsManager.java:80: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		list.set(0, (Object)(new Integer(i).toString()));
    [javac] 		                     ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/preferences/implementations/ShowAgainDialog.java:159: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]         		showAgainPref.setValue(new Boolean(!showAgainBox.isSelected()));
    [javac]         		                       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/preferences/implementations/ShowAgainDialog.java:166: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]         		showAgainPref.setValue(new Boolean(!showAgainBox.isSelected()));
    [javac]         		                       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/preferences/implementations/ShowAgainDialog.java:183: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]             	showAgainPref.setValue(new Boolean(!showAgainBox.isSelected()));
    [javac]             	                       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LinkToolPanel.java:170: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		i[0]= new Integer(0);
    [javac] 		      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LinkToolPanel.java:171: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		i[1]= new Integer(1);
    [javac] 		      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LinkToolPanel.java:172: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		i[2]= new Integer(2);
    [javac] 		      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/LinkToolPanel.java:173: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		i[3]= new Integer(3);
    [javac] 		      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ArrowToolPanel.java:139: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		i[0]= new Integer(0);
    [javac] 		      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ArrowToolPanel.java:140: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac] 		i[1]= new Integer(1);
    [javac] 		      ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/FontEditorPanel.java:574: warning: [removal] Integer(String) in Integer has been deprecated and marked for removal
    [javac]                 	return new Integer(mSizeField.getSelectedItem().toString()); 
    [javac]                 	       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/gui/StrokeMenuButton.java:49: warning: [removal] Float(float) in Float has been deprecated and marked for removal
    [javac]             values[i] = new Float(pValues[i]);
    [javac]                         ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/gui/StrokeMenuButton.java:55: warning: [removal] Float(float) in Float has been deprecated and marked for removal
    [javac]         mCurrentValue = new Float(1);
    [javac]                         ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/gui/StrokeMenuButton.java:59: warning: [removal] Float(float) in Float has been deprecated and marked for removal
    [javac]         mCurrentValue = new Float(1);
    [javac]                         ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/gui/StrokeMenuButton.java:65: warning: [removal] Float(float) in Float has been deprecated and marked for removal
    [javac]             mCurrentValue = new Float(1);
    [javac]                             ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VueUtil.java:684: warning: [removal] Byte(byte) in Byte has been deprecated and marked for removal
    [javac]                                + " " + pad(' ', 4, new Byte(b).toString())
    [javac]                                                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VueUtil.java:710: warning: [removal] Integer(int) in Integer has been deprecated and marked for removal
    [javac]                                + " " + pad(' ', 6, new Integer(cv).toString())
    [javac]                                                    ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/VueResources.java:741: warning: [removal] Integer(String) in Integer has been deprecated and marked for removal
    [javac]             Integer i = new Integer(s);
    [javac]                         ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/edu/tufts/vue/ui/ConfigurationUI.java:200: warning: [removal] Boolean(boolean) in Boolean has been deprecated and marked for removal
    [javac]                 Boolean isMandatory = new Boolean(true);
    [javac]                                       ^
    [javac] /home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ds/DataTree.java:1953: error: incompatible types: Vector<TreeNode> cannot be converted to Vector<DataNode>
    [javac]             return super.children;
    [javac]                         ^
    [javac] Note: Some input files use or override a deprecated API.
    [javac] Note: Recompile with -Xlint:deprecation for details.
    [javac] Note: Some input files additionally use or override a deprecated API that is marked for removal.
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 4 errors
    [javac] 100 warnings

**Comment by makingglitches** [comment]
Yeah Fedora 38 doesn't want to install openjdk 16, which is as I understand equivalent to the original version 1.6 you had in that variable.

**Comment by makingglitches** [comment]
Looking closer this really boils down to two errors:

/home/john/Downloads/VUE-master/VUE2/src/tufts/vue/Actions.java:1332: error: cannot find symbol
[javac] netscape.javascript.JSObject win = (netscape.javascript.JSObject) netscape.javascript.JSObject.getWindow(VueApplet.getInstance());

/home/john/Downloads/VUE-master/VUE2/src/tufts/vue/ds/DataTree.java:1953: error: incompatible types: Vector cannot be converted to Vector
[javac] return super.children;
[javac] ^

**Comment by makingglitches** [comment]
As Per ChatGPT when asked about it.

As of my knowledge cutoff date in January 2022, the `netscape.javascript.JSObject` class was part of the LiveConnect API used for communication between JavaScript and Java in web browsers. However, note that the LiveConnect API is considered outdated, and modern web development typically relies on more standardized and secure approaches.

If you are working with Java and JavaScript in a web environment, you might want to consider using the JavaFX WebView or, more commonly, integrating JavaScript with Java through the Java Scripting API (javax.script) or using frameworks like Nashorn, depending on your specific use case.

Keep in mind that information might have changed after my last update in January 2022, so it's a good idea to check the latest documentation or community discussions for any updates on this topic.

**Comment by makingglitches** [comment]
So its just a matter of determining what that was doing in the code.

**Comment by makingglitches** [comment]
I'm gonna look at the other but I'm guessing its mostly going to be related to namespace addressing.

**Comment by makingglitches** [comment]
Thanks so much for making me redo this, pretty sure it was fixed. Please get me away from the fucked up human traffickers in fucking Boulder Colorado.

**Comment by rpavlik** [comment]
If you look at my fork (see another PR) I have updated this substantially, but I have not heard from upstream on resolving some license issues I found, nor on merging any of my fixes.

**Comment by makingglitches** [comment]
@rpavlik in looking a bit closer the JSObject is trying to grab the window object of an applet. Could probably just drop this altogether.  Is your fork building ?

Unless I'm not understanding something would appear all calls to JObject are trying to get a DOM element for when this si likely embedded in a webpage for displaying a drawing.

The other is likely namespace. 

What does your fork modify ? I was actually considering forking myself, but if you already fixed this this time...

**Comment by makingglitches** [comment]
@rpavlik  define license issues if you would please.

**Comment by makingglitches** [comment]
HEHE see any of these yet ? :P 

![Screenshot from 2024-01-07 13-14-48](https://github.com/VUE/VUE/assets/53788764/f8dacb08-7d06-4165-8f04-03b80434e74d)
![Screenshot from 2024-01-07 13-26-35](https://github.com/VUE/VUE/assets/53788764/a400db3c-acf8-43de-a71a-b4aa48a4b5ed)

**Comment by makingglitches** [comment]
@rpavlik  unrelated but kind of related to code safety look at  #63

**Comment by makingglitches** [comment]
@rpavlik I was looking over your code and you know without a full code review or some good tools you can never really be sure, but I found these binaries they added to the project. kind of want to know what they do.

**Comment by makingglitches** [comment]
Fixed this. Please grant permission to  merge.

https://github.com/makingglitches/VUENoApplets

**Comment by rpavlik** [comment]
I doubt they'll give you permission to merge: I overhauled the whole build system, getting it to work with JDK 17, fixed a bunch of issues, and I haven't even heard a reply. Like I said, I fixed most of these things except the binaries for which there is no source, just class files, and the inherent incompatibility of GPL2 and Apache licenses.

More on license incompat here: https://github.com/VUE/VUE/issues/49

(I'm nobody special re: VUE - no association with Tufts, etc. I was just a happy user for a long time who wanted to keep it alive at least a little bit. Unfortunately the license thing means that I can't legally distribute binaries I make, so that took away much of my energy and enthusiasm for working on the project.)

**Comment by makingglitches** [comment]
Well there are issues I'm going to have to look into anyway. The entire file menu for example is disabled.  Which is weird so i'm gonna have to do a side by side compare in my fork and make sure i didn't remove anything vital.

**Comment by makingglitches** [comment]
Also the search box isn't working. strangely.

**Comment by makingglitches** [comment]
I honestly wouldn't be surprised if i already did this.

**Comment by makingglitches** [comment]
@rpavlik  I'm noticing 3.3.0 works as advertised....  but then there were 36 commits on top of that... i looked over the diffs of my fork and i'm not seeing what could have disabled the whole file menu. answer is nothing could have.... didn't even touch that section of the code, just removed things related to the applet.

**Comment by makingglitches** [comment]
@rpavlik and its using the awt interface for everything else..

**Comment by makingglitches** [comment]
@rpavlik and swing.

**Comment by sfraize** [comment]
As the implementor of the VUE's applet support, I can tell you it was very experimental and as you can see had already begun to be commented out via some constant branch conditionals.  I don't know if anyone ever actually used VUE as an applet, we just wanted to see how possible it would be (maybe Mike can correct  me on that).   The long and short of it though is should be fundamentally safe to remove applet support entirely -- it was patched in on top after most of the VUE app already existed. For minimal changes, I would start by replacing the body of VUE.isApplet() with "return false;" and see how far that gets you...

As for JSObject references in Actions.java (e.g., the ZoteroFFExtension) that looks applet related as well.  I do recall that
Zotero was (is?) a very big deal in academic circles and was very important to VUE, even if it was experimental we may have done something special with that within applet support.

I do want to say I'm disappointed to hear there are licensing issues?   As someone who was there, I can tell you that the Mellon Foundation, who solely funded all three of the grants at Tufts that produced VUE, wouldn't be happy to hear there are roadblocks to redistributing your own versions of VUE -- pushing the state of the art with VUE and supporting open source were part of the original idea from the get-go.  (As would have David Kahle, the principle investigator at Tufts that spearheaded the project).

Is it mainly the SimplyHTML mystery code issue?

**Comment by sfraize** [comment]
When I looked at upgrading VUE to JDK17 a couple of years ago, I only got as far as documenting that VUE should just settle for JDK16 instead, because the IllegalAccessException you're seeing at the top can be handled with "java --illegal-access=permit ..." in JDK16, but that option is was removed as of JDK17...   As for fixing the original illegal access, I've no idea if that's possible w/out replacing the entire XML parsing stack (not that I'd choose XML as a save format again).  I do recall us using 3rd party XML libraries because at the time Sun's included stuff was slow and lacking basic features, but in retrospect I wished we had found a way to keep in simpler.  The need for the XML mapping file added alot of complexity.

**Comment by rpavlik** [comment]
I have actually gotten it to build under JDK 17 and run just fine. Got the castor XML stuff straightened out, etc. It could use a pass with a code formatter, remove commented out code, etc. I already ran a number of semi-automated fixes from IDEA/Android Studio on it to help solve the JDK version issues. I have it building with Gradle, with some updated dependencies, and a number of bugs fixed. All the source is in my repo in clean commits. However...

The SimplyHTML2 library is GPL2.0+. There are no sources included, only a binary (direct violation of GPL2), there have been additional classes added to the jar without sources (just not good practice, and confusing from a license clearing perspective), the package depends on Apache-licensed dependencies. The only confirmed compatibility between GPL and Apache licenses is GPL 3 with Apache 2.0. This is GPL2+, which can upgrade to 3, but there are other files in VUE that are Apache-1.1 licensed. (At least the weird old apache-commons.jar, which contains a mix of old versions of Apache Commons libraries, apparently with some modifications. Replacing that jar has been very tricky.) FSF considers Apache licenses before 2.0 to be incompatible with all of the GPL versions.

(I am not a lawyer)

So it's not quite as bad as I had remembered, I thought SimplyHTML2 was GPL-2.0-only, which would be incompatible with even Apache-2.0. Looks like maybe, if we can either switch to a release of simplyhtml2 that has sources, and (this is the hard part) try to track down the sources for the Vue-specific files in this weird JAR of it, and go through and make sure nothing is GPL-2.0-only and all Apache-1.1 is replaced, it might be possible to clean it up enough to distribute it.

(It looks like there is some light maintenance of simplyhtml still taking place in the context of the FreePlane mindmapper here: https://github.com/freeplane/shtml )

Yeah it's a real drag when I found out about the license issues, more or less by accident when trying to update the build system, stop vendoring all the libraries in jar files, and update to modern dependencies. Really took a lot of my eagerness to help with it out, because I realized I couldn't safely share binaries, even though I had even figured out how to build binaries on GitHub. Any help you could offer in figuring out how to untangle the license mess would be appreciated.

**Comment by makingglitches** [comment]
@sfraize  I have been kind of busy but I found that getting it to build mainly involved removing applets and updating a few other utilities, but it still will require a little more work to get it back up completely, mostly it works, but there are a few issues to work out. If you look at my fork, it builds and mostly runs but it needs a bit of work, and  I didn't make that many modifications to the stopping point I modded years ago when moving towards serialization, so it shouldn't take that long. One problem is it does seem to have a number of unused and unfinished options that don't seem to do much or have much use in the tool itself.

**Comment by makingglitches** [comment]
@sfraize however considering the main purpose of the tool, the others work fine.
